examples/step-78/step-78.cc

```diff
@@ -0,0 +1,835 @@
+  {
+    (void)component;
+    return -std::pow(p(component), 2) + 6;
+    // Below is the original initial condition
+    // return std::max(p(component) - s_price, 0.);
+  }
+
+  // Boundary Conditions Functions
+
+  // Left Boundary Condition
+  template <int dim>
+  class LeftBoundaryValues : public Function<dim>
+  {
+  public:
+    LeftBoundaryValues();
+    virtual double value(const Point<dim> & p,
+                         const unsigned int component = 0) const override;
+  };
+
+  template <int dim>
+  double LeftBoundaryValues<dim>::value(const Point<dim> & p,
+                                        const unsigned int component) const
+  {
+    (void)component;
+    (void)p;
+    return -std::pow(this->get_time(), 2) + 6;
+
+    // Below is the original left boundary condition
+    // return 0.0;
+  }
+  // Right Boundary Condition
+  template <int dim>
+  class RightBoundaryValues : public Function<dim>
+  {
+  public:
+    RightBoundaryValues(double s_price, double r);
+    virtual double value(const Point<dim> & p,
+                         const unsigned int component = 0) const override;
+
+  private:
+    double s_price;
+    double r;
+  };
+
+
+  template <int dim>
+  double RightBoundaryValues<dim>::value(const Point<dim> & p,
+                                         const unsigned int component) const
+  {
+    (void)component;
+    return -std::pow(p(component), 2) - std::pow(this->get_time(), 2) + 6;
+    // Below is the original right boundary condition
+    // return (p(component) - _s_price) * exp((-_r) * (this->get_time()));
+  }
+
+  // Forcing Function (Right hand side function)
+  template <int dim>
+  class RightHandSide : public Function<dim>
+  {
+  public:
+    RightHandSide(double sigma, double r);
+    virtual double value(const Point<dim> & p,
+                         const unsigned int component = 0) const override;
+
+  private:
+    double sigma;
+    double r;
+  };
+
+  template <int dim>
+  double RightHandSide<dim>::value(const Point<dim> & p,
+                                   const unsigned int component) const
+  {
+    (void)component;
+    // return 0;
+    return 2 * (this->get_time()) - std::pow(sigma * p(component), 2) -
+           2 * r * std::pow(p(component), 2) -
+           r * (-std::pow(p(component), 2) - std::pow(this->get_time(), 2) + 6);
+
+    // Below is the original right hand side
+    // return 0.0
+  }
+
+  template <int dim>
+  InitialConditions<dim>::InitialConditions(double s_price)
+    : s_price(s_price)
+  {}
+
+  template <int dim>
+  LeftBoundaryValues<dim>::LeftBoundaryValues()
+  {}
+
+  template <int dim>
+  RightBoundaryValues<dim>::RightBoundaryValues(double s_price, double r)
+    : s_price(s_price)
+    , r(r)
+  {}
+
+  template <int dim>
+  RightHandSide<dim>::RightHandSide(double sigma, double r)
+    : sigma(sigma)
+    , r(r)
+  {}
+
+  template <int dim>
+  Solution<dim>::Solution(double maturity_time)
+    : maturity_time(maturity_time)
+  {}
```

Comment:
I'm not sure what you mean here with the mixing of member functions. I am defining some member variables that have the same name such as 'r'. Is this what you are referencing to?

Reply:
These are all definition of constructors for different classes in lines 283 to 307.

For tutorials, we made sure that all definitions of class member functions immediately follow the class definition. So you basically have one block of code that deals with that particular class and no other to prevent confusion amongst readers. This way we do not mix up implementations for different classes.

Just move the definitions to the correct position. I mean something like this:
```
// Solution
template <int dim> class Solution {...};               // declaration of class
template <int dim> Solution<dim>::Solution(...){...}   // definition of constructor
template <int dim> Solution<dim>::value(...){...}      // definition of another member
...

// InitialConditions
template <int dim> class InitialConditions {...};
template <int dim> InitialConditions<dim>::InitialConditions(...){...}
...
```